app/models/metadata.rb

```diff
@@ -21,4 +21,9 @@ def get_records(descending = true)
   def self.daac_from_concept_id(concept_id)
     concept_id.partition('-').last
   end
+
+  def self.extract_campaign(collection_data)
+    keyword = RecordHelper::CAMPAIGN_COLUMNS.select { |column| collection_data.keys.include?(column)}
```

Comment:
hmm, my idea might not working because you are extracting this at the metadata level and not at the record level but worth thinking about if it is possible to use that record_formats if possible.